scss/themes/native.scss

```diff
@@ -0,0 +1,124 @@
+$main-theme-color: -moz-field;
+$body-bg-color: -moz-dialog;
+$anchor-color: -moz-nativehyperlinktext;
+$main-text-color: -moz-fieldtext;
+$border-color: -moz-CellHighlight;
+$main-bg-color: -moz-dialog;
+$secondary-text-color: -moz-fieldtext;
+$toast-border: $main-text-color;
+$toast-bg: -moz-CellHighlight;
+$focus-outline: $main-theme-color;
+$compose-background: $main-theme-color;
+$deemphasized-color: -moz-fieldtext;
+
+$scrollbar-face: #{$secondary-text-color} !default;
+$scrollbar-face-hover: #{$secondary-text-color} !default;
+$scrollbar-face-active: #{$secondary-text-color} !default;
+$scrollbar-track: #{$body-bg-color} !default;
+
+:root {
+    --button-primary-bg: $main-theme-color;
```

Comment:
The issue is that we don't know what kind of transformation to apply (we can't hardcode `lighten` or `darken` for certain variables). We could try to infer if the `-moz` colors are dark/light and switch the transform based on that but that would be britle.

Reply:
Hmmmm I think ultimately for this feature to work, I would want us to be able to apply the `lighten()` and `darken()` functions (or something similar). We could just hard-code the colors in advance, but I agree that would be brittle.

Maybe it's time I move from SCSS to something that can be run in a lightweight way on the client side. As I mentioned, this would also fix #278.

Reply:
https://github.com/bgrins/TinyColor/ and https://github.com/Qix-/color/blob/ seem to offer the required functionality.